packages/nest/src/feature.decorator.ts

```diff
@@ -1,62 +1,52 @@
-import { createParamDecorator, ExecutionContext, Inject } from '@nestjs/common';
+import { createParamDecorator, Inject } from '@nestjs/common';
 import { EvaluationContext, FlagValue, JsonValue, OpenFeature } from '@openfeature/server-sdk';
 import { getOpenFeatureClientToken } from './open-feature.module';
 import { from } from 'rxjs';
+import { OpenFeatureContextService } from './feature.service';
+import { SharedAsyncLocalStorage } from './async-storage';
 
 interface FeatureClientProps {
   name?: string;
 }
 
 export const FeatureClient = (props?: FeatureClientProps) => Inject(getOpenFeatureClientToken(props?.name));
+export const FeatureContextService = () => Inject(OpenFeatureContextService);
 
 interface FeatureProps<T extends FlagValue> {
   clientName?: string;
   flagKey: string;
   defaultValue: T;
   context?: EvaluationContext;
-  contextFactory?: (executionContext: ExecutionContext) => EvaluationContext | undefined;
 }
 
 function getClientForEvaluation(clientName?: string, context?: EvaluationContext) {
   return clientName ? OpenFeature.getClient(clientName, context) : OpenFeature.getClient(context);
 }
 
 export const BooleanFeatureFlag = createParamDecorator(
-  (
-    { clientName, flagKey, defaultValue, context, contextFactory }: FeatureProps<boolean>,
-    executionContext: ExecutionContext,
-  ) => {
-    const client = getClientForEvaluation(clientName, context);
-    return from(client.getBooleanDetails(flagKey, defaultValue, contextFactory?.(executionContext)));
+  ({ clientName, flagKey, defaultValue }: FeatureProps<boolean>) => {
+    const client = getClientForEvaluation(clientName);
+    return from(client.getBooleanDetails(flagKey, defaultValue, SharedAsyncLocalStorage.getStore()?.context));
   },
 );
 
 export const StringFeatureFlag = createParamDecorator(
-  (
-    { clientName, flagKey, defaultValue, context, contextFactory }: FeatureProps<string>,
-    executionContext: ExecutionContext,
-  ) => {
+  ({ clientName, flagKey, defaultValue, context }: FeatureProps<string>) => {
     const client = getClientForEvaluation(clientName, context);
-    return from(client.getStringDetails(flagKey, defaultValue, contextFactory?.(executionContext)));
+    return from(client.getStringDetails(flagKey, defaultValue, SharedAsyncLocalStorage.getStore()?.context));
```

Comment:
You could use the `setTransactionContextPropagator` to automatically extract transaction scoped context within the SDK itself. You can see an example in the playground.

https://github.com/open-feature/playground/blob/main/packages/app/src/app/app.module.ts#L46

You'll just need to create a context propagator [like this](https://github.com/open-feature/playground/blob/main/packages/openfeature-extra/src/lib/transaction-context/async-local-storage.ts).

Reply:
Oh ya this could be a cool idea!
I thought about it, but thought that this API is not continued.
I will make a PR using this tomorrow or so to try I really like it!

Reply:
It's marked as experimental but we could remove that disclaiming if we start using it in this SDK.

Reply:
Cool, I will give it a try and add it tomorrow or so🙃

Reply:
This looks great!

[It's merging context](https://github.com/open-feature/js-sdk/blob/main/packages/server/src/client/open-feature-client.ts#L202) so, [this PR](https://github.com/open-feature/js-sdk/pull/738) is not necessary :rocket:

Reply:
@beeme1mr @luizgribeiro 
I changed to choose the context propagator for all context propagation.
This makes the implementation way cleaner, I really like it. 
I also removed the ContextService as I think this would not be needed in this implementation.
What do you think?

Reply:
It looks good so far.